splipy/BSplineBasis.py

```diff
@@ -446,6 +447,23 @@ def matches(self, bspline, reverse=False):
                                 (bspline.knots-bspline.knots[0]) / dt2,
                                 atol=state.knot_tolerance)
 
+    def snap(self, t):
+        """  Snap evaluation points to knots if they are sufficiently close
+        as given in by state.state.knot_tolerance. This will modify the input vector t
+
+        :param t: evaluation points
+        :type  t: [float]
+        :return: none
+        """
+        n = len(self.knots)
+        for j in range(len(t)):
```

Comment:
If the only purpose of `j` is to index into `t` then just `for tt in t` or similar.

Reply:
I also need to change the content of `t` (see line 462) and then your proposed syntax won't work.